routes/truckRoutes.js

```diff
@@ -1,2 +1,83 @@
-// Connect to mongodb in this module as this is where you'll be making creat/read/delete calls to your database
-// use 'mongodb://localhost/foodTruckAPI' for your mongoose connection string
\ No newline at end of file
+/*eslint-env node*/
+
+//express
+var express = require('express');
+
+//router
+var router = express.Router();
+
+//the food trucks
+var foodTruck = require('../models/truckModel');
+
+router.route('/')
+.get(function (request, response) {
+
+    //return all the trucks
+  foodTruck.find().exec(function (err, result) {
+
+    if (err) {
+
+            //sorry error getting trucks
+        response.status(500).send(err);
+      } else {
+
+            //yay saved
+        response.send(result);
+      }
+
+  });
+})
+
+.post(function (request, response) {
+
+    //create the new truck
+  var newFoodTruck = new foodTruck(request.body);
+
+    //save it
+  newFoodTruck.save(function (err) {
+    if (err) {
+
+            //sorry error saving
+        response.status(500).send(err);
+      } else {
+
+            //yay saved
+        response.status(201).send(newFoodTruck);
+      }
+  });
+});
+
+router.route('/:name')
+.get(function (request, response) {
+
+    //find the food truck, changing to _id (used to be name)
+  foodTruck.find({ _id: request.params.name }, function (err, result) {
+    if (err) {
+
+            //sorry no truck
+        response.status(404).json('no truck found: ' + request.params.name);
+      } else {
+
+            //yay found truck
+        response.send(result);
+      }
+  });
+})
+
+.delete(function (request, response) {
+
+  foodTruck.findOneAndRemove({ name: request.params.name }, function(err){
```

Comment:
This doesn't work. The parameter being sent is an id, not a name, which means that after you "delete" it, it will show up again when you refresh the page because it won't find a name that matches that id. The method you want to use is find ByIdAndRemove.